## Capstone draft 1. Getting familiar with the data

### Plan of attack

1. Read the file (Acquire data)
2. Take an initial look at the data (including the head, tail, info, etc.) 
3. Wrangle, process, and clean the data 
4. EDA - Exploratory Data Analysis

### Table of Contents
- [Importing and initial reading](#Part-1a)
- [Shape and dictionary information](#Part-1b)
- [Data cleaning](#Part-2)
- [Obeservations](#Part-3)

## Part 1a
importing and initial reading of each individual CSV

In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
mer = pd.read_csv('Merged.csv')
mer

County  Year  Population   Males  Females  White Males  \
0      Alameda    2001     1468652  723957   744695       430526   
1      Alameda    2002     1460438  720008   740430       426013   
2      Alameda    2003     1451418  715373   736045       421147   
3      Alameda    2004     1441496  710595   730901       416592   
4      Alameda    2005     1435870  708070   727800       413816   
...          ...   ...         ...     ...      ...          ...   
62519     Yuba    2018       77557   39249    38308        30806   
62520     Yuba    2019       78668   39819    38849        31245   
62521     Yuba    2020       81994   42004    39990        32575   
62522     Yuba    2021       83216   42550    40666        32760   
62523     Yuba    2022       84310   43017    41293        32745   

       White Females  Black Males  Black Females  Native Males  ...  \
0             426851       101773         116649          5637  ...   
1             422003        98418         113065          5651  ...   
2             417397        95566         110066          5691  ...   
3             412797        93024         107047          5729  ...   
4             409781        90566         104032          5748  ...   
...              ...          ...            ...           ...  ...   
62519          30125         1861           1521          1050  ...   
62520          30419         1909           1595          1029  ...   
62521          31007         2083           1714          1162  ...   
62522          31234         2200           1826          1200  ...   
62523          31381         2292           1985          1205  ...   

       Hispanic-White Mixed Males  Hispanic-White Mixed Females  \
0                          137800                        125878   
1                          139729                        127753   
2                          140761                        128789   
3                          141652                        130010   
4                          143345                        131666   
...                           ...                           ...   
62519                       10162                          9668   
62520                       10462                         10024   
62521                       11102                         10410   
62522                       11487                         10788   
62523                       11981                         11247   

       Hispanic-Black Mixed Males  Hispanic-Black Mixed Females  \
0                            5423                          5630   
1                            5400                          5632   
2                            5389                          5664   
3                            5360                          5661   
4                            5386                          5651   
...                           ...                           ...   
62519                         486                           469   
62520                         493                           481   
62521                         574                           563   
62522                         576                           586   
62523                         617                           591   

       Hispanic-Native Mixed Males  Hispanic-Native Mixed Females  \
0                             4401                           4138   
1                             4448                           4162   
2                             4457                           4131   
3                             4469                           4146   
4                             4429                           4150   
...                            ...                            ...   
62519                          949                            982   
62520                          951                            973   
62521                         1035                           1006   
62522                         1088      

In [3]:
raw_inf = pd.read_csv('infectious1.csv')
raw_inf

Disease   County  Year     Sex  Cases  Population Rate
0               Anaplasmosis  Alameda  2001  Female      0      746596    -
1               Anaplasmosis  Alameda  2001    Male      0      718968    -
2               Anaplasmosis  Alameda  2001   Total      0     1465564    -
3               Anaplasmosis  Alameda  2002  Female      0      747987    -
4               Anaplasmosis  Alameda  2002    Male      0      720481    -
...                      ...      ...   ...     ...    ...         ...  ...
189916  Zika Virus Infection     Yuba  2021    Male      0       39302    -
189917  Zika Virus Infection     Yuba  2021   Total      0       82211    -
189918  Zika Virus Infection     Yuba  2022  Female      0       39108    -
189919  Zika Virus Infection     Yuba  2022    Male      0       39558    -
189920  Zika Virus Infection     Yuba  2022   Total      0       82533    -

[189921 rows x 7 columns]

In [4]:
raw_inf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 189921 entries, 0 to 189920
Data columns (total 7 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   Disease     189921 non-null  object
 1   County      189921 non-null  object
 2   Year        189921 non-null  int64 
 3   Sex         189921 non-null  object
 4   Cases       189921 non-null  int64 
 5   Population  189921 non-null  int64 
 6   Rate        189921 non-null  object
dtypes: int64(3), object(4)
memory usage: 10.1+ MB


## Part 1b 
Shape and dictionary information of each CSV

In [5]:
raw_inf.shape

(189921, 7)

In [6]:
mer.shape

(62524, 75)

In [7]:
raw_inf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 189921 entries, 0 to 189920
Data columns (total 7 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   Disease     189921 non-null  object
 1   County      189921 non-null  object
 2   Year        189921 non-null  int64 
 3   Sex         189921 non-null  object
 4   Cases       189921 non-null  int64 
 5   Population  189921 non-null  int64 
 6   Rate        189921 non-null  object
dtypes: int64(3), object(4)
memory usage: 10.1+ MB


| Column Name | Datatype |   |
|-------------|----------|---|
| Disease     | Object   |   |
| County      | Object   |   |
| Year        | int64    |   |
| Sex         | Object   |   |
| Cases       | int64    |   |
| Population  | int64    |   |
| Rate        | Object   |   |
|             |          |   |
|             |          |   |

In [8]:
mer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62524 entries, 0 to 62523
Data columns (total 75 columns):
 #   Column                                Non-Null Count  Dtype 
---  ------                                --------------  ----- 
 0   County                                62524 non-null  object
 1   Year                                  62524 non-null  int64 
 2   Population                            62524 non-null  int64 
 3   Males                                 62524 non-null  int64 
 4   Females                               62524 non-null  int64 
 5   White Males                           62524 non-null  int64 
 6   White Females                         62524 non-null  int64 
 7   Black Males                           62524 non-null  int64 
 8   Black Females                         62524 non-null  int64 
 9   Native Males                          62524 non-null  int64 
 10  Native Females                        62524 non-null  int64 
 11  Asian Males                 

| Column Name                       | Data type |
|-----------------------------------|-----------|
| County                            | Object    |
| Year                              | int       |
| Population                        | int       |
| Males                             | int       |
| Females                           | int       |
| White Males                       | int       |
| White Females                     | int       |
| Black Males                       | int       |
| Black Females                     | int       |
| Native Males                      | int       |
| Native Females                    | int       |
| Asian Males                       | int       |
| Asian Females                     | int       |
| Hawaiian Male                     | int       |
| Hawaiian Females                  | int       |
| Multi-Race Males                  | int       |
| Multi-Race Females                | int       |
| White-Mixed Males                 | int       |
| White-Mixed Females               | int       |
| Black-Mixed Males                 | int       |
| Black-Mixed Females               | int       |
| Native-Mixed Males                | int       |
| Native-Mixed Females              | int       |
| Asian-Mixed Males                 | int       |
| Asian-Mixed Females               | int       |
| Hawaiian-Mixed Males              | int       |
| Hawaiian-Mixed Females            | int       |
| Not-Hispanic Males                | int       |
| Not-Hispanic Females              | int       |
| Not-Hispanic White Males          | int       |
| Not-Hispanic White Females        | int       |
| Not-Hispanic Black Males          | int       |
| Not-Hispanic Black Males.1        | int       |
| Not-Hispanic Native Male          | int       |
| Not-Hispanic Native Female        | int       |
| Not-Hispanic Asian Male           | int       |
| Not-Hispanic Asian Female         | int       |
| Not-Hispanic Hawaiian Male        | int       |
| Not-Hispanic Hawaiian Female      | int       |
| Not_Hispanic Mixed Males          | int       |
| Not_Hispanic Mixed Females        | int       |
| Not_Hispanic White-Mixed Males    | int       |
| Not_Hispanic White-Mixed Females  | int       |
| Not_Hispanic Black-Mixed Males    | int       |
| Not_Hispanic Black-Mixed Males.1  | int       |
| Not_Hispanic Native-Mixed Males   | int       |
| Not_Hispanic Native-Mixed Females | int       |
| Not_Hispanic Asian-Mixed Males    | int       |
| Not_Hispanic Asian-               | int       |

## Part 2
Data cleaning of each CSV

In [9]:
raw_inf.duplicated().sum()

0

In [10]:
raw_inf.isna().sum()

Disease       0
County        0
Year          0
Sex           0
Cases         0
Population    0
Rate          0
dtype: int64

In [11]:
mer.duplicated().sum()

61248

In [12]:
mer.isna().sum()

County                             0
Year                               0
Population                         0
Males                              0
Females                            0
                                  ..
Hispanic-Native Mixed Females      0
Hispanic-Asian Mixed Males         0
Hispanic-Asian Mixed Females       0
Hispanic-Hawaiian Mixed Males      0
Hispanic-Hawaiian Mixed Females    0
Length: 75, dtype: int64

We don't have any duplicates or null values in either of our datasets. That makes our job way easier. Now we can focus on the other parts of EDA

In [13]:
# Using T to transpose to make the original columns and rows switch, then reapplying .duplicated()
raw_inf.T.duplicated()

Disease       False
County        False
Year          False
Sex           False
Cases         False
Population    False
Rate          False
dtype: bool

In [14]:
mer.T.duplicated()

County                             False
Year                               False
Population                         False
Males                              False
Females                            False
                                   ...  
Hispanic-Native Mixed Females      False
Hispanic-Asian Mixed Males         False
Hispanic-Asian Mixed Females       False
Hispanic-Hawaiian Mixed Males      False
Hispanic-Hawaiian Mixed Females    False
Length: 75, dtype: bool

Again, there are duplicated present in either dataset. 

In [15]:
raw_inf.isna().sum(axis=0)

Disease       0
County        0
Year          0
Sex           0
Cases         0
Population    0
Rate          0
dtype: int64

In [16]:
mer.isna().sum(axis=0)

County                             0
Year                               0
Population                         0
Males                              0
Females                            0
                                  ..
Hispanic-Native Mixed Females      0
Hispanic-Asian Mixed Males         0
Hispanic-Asian Mixed Females       0
Hispanic-Hawaiian Mixed Males      0
Hispanic-Hawaiian Mixed Females    0
Length: 75, dtype: int64

In [17]:
mer.describe()

Year    Population         Males       Females   White Males  \
count  62524.00000  6.252400e+04  6.252400e+04  6.252400e+04  6.252400e+04   
mean    2011.50000  6.467946e+05  3.224646e+05  3.243301e+05  2.416891e+05   
std        6.34434  1.416602e+06  7.019549e+05  7.147048e+05  5.158111e+05   
min     2001.00000  1.042000e+03  5.520000e+02  4.820000e+02  3.730000e+02   
25%     2006.00000  4.543750e+04  2.275075e+04  2.272725e+04  2.083225e+04   
50%     2011.50000  1.798945e+05  9.000750e+04  9.075600e+04  7.847150e+04   
75%     2017.00000  6.875930e+05  3.423612e+05  3.370392e+05  2.308010e+05   
max     2022.00000  1.010571e+07  4.980981e+06  5.125335e+06  3.660384e+06   

       White Females    Black Males  Black Females  Native Males  \
count   6.252400e+04   62524.000000   62524.000000  62524.000000   
mean    2.390493e+05   21143.731191   21637.058777   4865.760188   
std     5.117573e+05   60197.115094   66947.154083   9550.713666   
min     3.180000e+02       0.000000       0.000000     27.000000   
25%     2.086925e+04     438.750000     196.750000    676.000000   
50%     8.050250e+04    2310.000000    1651.500000   1940.000000   
75%     2.219042e+05   11322.250000   11079.750000   4879.500000   
max     3.617533e+06  456322.000000  520700.000000  75913.000000   

       Native Females  ...  Hispanic-White Mixed Males  \
count    62524.000000  ...                6.252400e+04   
mean      4697.060345  ...                1.147597e+05   
std       9144.823030  ...                3.067983e+05   
min         23.000000  ...                2.900000e+01   
25%        608.500000  ...                4.203250e+03   
50%       1932.500000  ...                2.197600e+04   
75%       4664.750000  ...                1.063122e+05   
max      71690.000000  ...                2.288151e+06   

       Hispanic-White Mixed Females  Hispanic-Black Mixed Males  \
count                  6.252400e+04                62524.000000   
mean                   1.112514e+05                 3088.554075   
std                    3.041112e+05                 7940.348407   
min                    1.800000e+01                    1.000000   
25%                    3.708250e+03                   83.750000   
50%                    1.982250e+04                  438.000000   
75%                    9.881450e+04                 2686.250000   
max                    2.300374e+06                68266.000000   

       Hispanic-Black Mixed Females  Hispanic-Native Mixed Males  \
count                   62524.00000                 62524.000000   
mean                     3192.87931                  4694.674765   
std                      8295.09291                 10494.489131   
min                         0.00000                     6.000000   
25%                        66.75000                   336.750000   
50%                       469.00000                  1252.500000   
75%                      2750.50000                  4518.250000   
max                     70097.00000                 89013.000000   

       Hispanic-Native Mixed Females  Hispanic-Asian Mixed Males  \
count                   62524.000000                62524.000000   
mean                     4496.130094                 2674.873041   
std                     10050.415631                 6130.827263   
min                         4.000000                    0.000000   
25%                       295.000000                   64.750000   
50%                      1207.500000                  443.500000   
75%                      4178.250000                 2655.250000   
max                     84397.000000                52481.000000   

       Hispanic-Asian Mixed Females  Hispanic-Hawaiian Mixed Males  \
count                  62524.000000                   62524.000000   
mean                    2717.501567                     656.739028   
std                     6210.075293                    1449.767961   
min                        0.000000                       0.00000

In [18]:
raw_inf.describe()

Year          Cases    Population
count  189921.000000  189921.000000  1.899210e+05
mean     2011.803355      10.690319  8.533487e+05
std         6.350271     146.497169  3.544983e+06
min      2001.000000       0.000000  5.480000e+02
25%      2006.000000       0.000000  2.938700e+04
50%      2012.000000       0.000000  1.255400e+05
75%      2017.000000       0.000000  4.246320e+05
max      2022.000000   10016.000000  3.952957e+07

In [19]:
raw_inf.select_dtypes(['object', 'int']).columns

Index(['Disease', 'County', 'Year', 'Sex', 'Cases', 'Population', 'Rate'], dtype='object')

In [20]:
mer.select_dtypes(['object', 'int']).columns

Index(['County', 'Year', 'Population', 'Males', 'Females', 'White Males',
       'White Females', 'Black Males', 'Black Females', 'Native Males',
       'Native Females', 'Asian Males', 'Asian Females', 'Hawaiian Male',
       'Hawaiian Females', 'Multi-Race Males', 'Multi-Race Females',
       'White-Mixed Males', 'White-Mixed Females', 'Black-Mixed Males',
       'Black-Mixed Females', 'Native-Mixed Males', 'Native-Mixed Females',
       'Asian-Mixed Males', 'Asian-Mixed Females', 'Hawaiian-Mixed Males',
       'Hawaiian-Mixed Females', 'Not-Hispanic Males', 'Not-Hispanic Females',
       'Not-Hispanic White Males', 'Not-Hispanic White Females',
       'Not-Hispanic Black Males', 'Not-Hispanic Black Males.1',
       'Not-Hispanic Native Male', 'Not-Hispanic Native Female',
       'Not-Hispanic Asian Male', 'Not-Hispanic Asian Female',
       'Not-Hispanic Hawaiian Male', 'Not-Hispanic Hawaiian Female',
       'Not_Hispanic Mixed Males', 'Not_Hispanic Mixed Females',
       ' Not_Hisp

#### `Clean_inf` is the newest version of `inf`. In this dataframe, I have dropped `Sex` and `Population` so that I can use the same values from mer instead. This way it will be easier to merge both CSVs as there won't be conflicting values. In dropping the other categories, the shape of the df is now (63307, 5) which is a drastic change from the (189921, 7) it previously was. 

In [21]:
clean_inf = pd.read_csv('new_inf.csv')

In [22]:
clean_inf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50044 entries, 0 to 50043
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   County   50044 non-null  object
 1   Year     50044 non-null  int64 
 2   Disease  50044 non-null  object
 3   Cases    50044 non-null  int64 
 4   Rate     50044 non-null  object
dtypes: int64(2), object(3)
memory usage: 1.9+ MB


In [23]:
clean_inf.shape

(50044, 5)

In [24]:
clean_inf.duplicated().sum()

0

In [25]:
clean_inf.isna().sum(axis = 0)

County     0
Year       0
Disease    0
Cases      0
Rate       0
dtype: int64

In [26]:
clean_inf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50044 entries, 0 to 50043
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   County   50044 non-null  object
 1   Year     50044 non-null  int64 
 2   Disease  50044 non-null  object
 3   Cases    50044 non-null  int64 
 4   Rate     50044 non-null  object
dtypes: int64(2), object(3)
memory usage: 1.9+ MB


In [27]:
mer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62524 entries, 0 to 62523
Data columns (total 75 columns):
 #   Column                                Non-Null Count  Dtype 
---  ------                                --------------  ----- 
 0   County                                62524 non-null  object
 1   Year                                  62524 non-null  int64 
 2   Population                            62524 non-null  int64 
 3   Males                                 62524 non-null  int64 
 4   Females                               62524 non-null  int64 
 5   White Males                           62524 non-null  int64 
 6   White Females                         62524 non-null  int64 
 7   Black Males                           62524 non-null  int64 
 8   Black Females                         62524 non-null  int64 
 9   Native Males                          62524 non-null  int64 
 10  Native Females                        62524 non-null  int64 
 11  Asian Males                 

## Part 3
Observations and Data Preprocessing of both CSVs combined.

#### `Final` is the latest version as it combines both the `Infectious` and `Merged` tables. I took the `Clean_inf` table from above and joined it to the demographics dataset. The final shape is (62234, 78). The good thing is I won't need to drop any additional fields as there are no null values or duplicates.


In [30]:
Final = pd.read_csv('final_merge.csv')
Final


Disease   County  Year  Cases Rate    Total   Males  \
0              Anaplasmosis  Alameda  2001      0    -  1468652  723957   
1              Anaplasmosis  Alameda  2002      0    -  1460438  720008   
2              Anaplasmosis  Alameda  2003      0    -  1451418  715373   
3              Anaplasmosis  Alameda  2004      0    -  1441496  710595   
4              Anaplasmosis  Alameda  2005      0    -  1435870  708070   
...                     ...      ...   ...    ...  ...      ...     ...   
62229  Zika Virus Infection     Yuba  2018      0    -    77557   39249   
62230  Zika Virus Infection     Yuba  2019      0    -    78668   39819   
62231  Zika Virus Infection     Yuba  2020      0    -    81994   42004   
62232  Zika Virus Infection     Yuba  2021      0    -    83216   42550   
62233  Zika Virus Infection     Yuba  2022      0    -    84310   43017   

       Females  White Males  White Females  ...  Hispanic-White Mixed Males  \
0       744695       430526         426851  ...                      137800   
1       740430       426013         422003  ...                      139729   
2       736045       421147         417397  ...                      140761   
3       730901       416592         412797  ...                      141652   
4       727800       413816         409781  ...                      143345   
...        ...          ...            ...  ...                         ...   
62229    38308        30806          30125  ...                       10162   
62230    38849        31245          30419  ...                       10462   
62231    39990        32575          31007  ...                       11102   
62232    40666        32760          31234  ...                       11487   
62233    41293        32745          31381  ...                       11981   

       Hispanic-White Mixed Females  Hispanic-Black Mixed Males  \
0                            125878                        5423   
1                            127753                        5400   
2                            128789                        5389   
3                            130010                        5360   
4                            131666                        5386   
...                             ...                         ...   
62229                          9668                         486   
62230                         10024                         493   
62231                         10410                         574   
62232                         10788                         576   
62233                         11247                         617   

       Hispanic-Black Mixed Females  Hispanic-Native Mixed Males  \
0                              5630                         4401   
1                              5632                         4448   
2                              5664                         4457   
3                              5661                         4469   
4                              5651                         4429   
...                             ...                          ...   
62229                           469                          949   
62230                           481                          951   
62231                           563                         1035   
62232                           586                         1088   
62233                           591                         1149   

       Hispanic-Native Mixed Females  Hispanic-Asian Mixed Males  \
0                               4138                        3883   
1                               4162                        3955   
2                               4131                        3984   
3                               4146                        4017   
4                               4150                        4109   
...                              ...                         ...   
62229                            982                         399   
6223

In [32]:
Final.shape

(62234, 78)

In [33]:
Final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62234 entries, 0 to 62233
Data columns (total 78 columns):
 #   Column                                Non-Null Count  Dtype 
---  ------                                --------------  ----- 
 0   Disease                               62234 non-null  object
 1   County                                62234 non-null  object
 2   Year                                  62234 non-null  int64 
 3   Cases                                 62234 non-null  int64 
 4   Rate                                  62234 non-null  object
 5   Total                                 62234 non-null  int64 
 6   Males                                 62234 non-null  int64 
 7   Females                               62234 non-null  int64 
 8   White Males                           62234 non-null  int64 
 9   White Females                         62234 non-null  int64 
 10  Black Males                           62234 non-null  int64 
 11  Black Females               

In [36]:
Final.duplicated().sum()

0

In [40]:
Final.T.duplicated().sum()

0

#### The table below shows all the elements in the dataset. Keep in mind that the datatypes are dynamic and most likely will change throughout the process. 

| Column Name                         | Data Type |
|-------------------------------------|-----------|
| Disease                             | Object    |
| County                              | Object    |
| Year                                | Int64     |
| Cases                               | Int64     |
| Rate                                | Object    |
| Total                               | Int64     |
| Males                               | Int64     |
| Females                             | Int64     |
| White Males                         | Int64     |
| White Females                       | Int64     |
| Black Males                         | Int64     |
| Black Females                       | Int64     |
| Native Males                        | Int64     |
| Native Females                      | Int64     |
| Asian Males                         | Int64     |
| Asian Females                       | Int64     |
| Hawaiian Male                       | Int64     |
| Hawaiian Females                    | Int64     |
| Multi-Race Males                    | Int64     |
| Multi-Race Females                  | Int64     |
| White-Mixed Males                   | Int64     |
| White-Mixed Females                 | Int64     |
| Black-Mixed Males                   | Int64     |
| Black-Mixed Females                 | Int64     |
| Native-Mixed Males                  | Int64     |
| Native-Mixed Females                | Int64     |
| Asian-Mixed Males                   | Int64     |
| Asian-Mixed Females                 | Int64     |
| Hawaiian-Mixed Males                | Int64     |
| Hawaiian-Mixed Females              | Int64     |
| Not-Hispanic Males                  | Int64     |
| Not-Hispanic Females                | Int64     |
| Not-Hispanic White Males            | Int64     |
| Not-Hispanic White Females          | Int64     |
| Not-Hispanic Black Males            | Int64     |
| Not-Hispanic Black Males.1          | Int64     |
| Not-Hispanic Native Male            | Int64     |
| Not-Hispanic Native Female          | Int64     |
| Not-Hispanic Asian Male             | Int64     |
| Not-Hispanic Asian Female           | Int64     |
| Not-Hispanic Hawaiian Male          | Int64     |
| Not-Hispanic Hawaiian Female        | Int64     |
| Not_Hispanic Mixed Males            | Int64     |
| Not_Hispanic Mixed Females          | Int64     |
| Not_Hispanic White-Mixed Males      | Int64     |
| Not_Hispanic White-Mixed Females    | Int64     |
| Not_Hispanic Black-Mixed Males      | Int64     |
| Not_Hispanic Black-Mixed Males.1    | Int64     |
| Not_Hispanic Native-Mixed Males     | Int64     |
| Not_Hispanic Native-Mixed Females   | Int64     |
| Not_Hispanic Asian-Mixed Males      | Int64     |
| Not_Hispanic Asian-Mixed Females    | Int64     |
| Not_Hispanic Hawaiian-Mixed Males   | Int64     |
| Not_Hispanic Hawaiian-Mixed Females | Int64     |
| Hispanic Males                      | Int64     |
| Hispanic Females                    | Int64     |
| Hispanic-White Males                | Int64     |
| Hispanic-White Females              | Int64     |
| Hispanic-Black Males                | Int64     |
| Hispanic-Black Females              | Int64     |
| Hispanic-Native Males               | Int64     |
| Hispanic-Native Females             | Int64     |
| Hispanic-Asian Males                | Int64     |
| Hispanic-Asian Females              | Int64     |
| Hispanic-Hawaiian Males             | Int64     |
| Hispanic-Hawaiian Females           | Int64     |
| Hispanic-Mixed Males                | Int64     |
| Hispanic-Mixed Females              | Int64     |
| Hispanic-White Mixed Males          | Int64     |
| Hispanic-White Mixed Females        | Int64     |
| Hispanic-Black Mixed Males          | Int64     |
| Hispanic-Black Mixed Females        | Int64     |
| Hispanic-Native Mixed Males         | Int64     |
| Hispanic-Native Mixed Females       | Int64     |
| Hispanic-Asian Mixed Males          | Int64     |
| Hispanic-Asian Mixed Females        | Int64     |
| Hispanic-Hawaiian Mixed Males       | Int64     |
| Hispanic-Hawaiian Mixed Females     | Int64     |

In [38]:
Final.isna().sum(axis = 0)

Disease                            0
County                             0
Year                               0
Cases                              0
Rate                               0
                                  ..
Hispanic-Native Mixed Females      0
Hispanic-Asian Mixed Males         0
Hispanic-Asian Mixed Females       0
Hispanic-Hawaiian Mixed Males      0
Hispanic-Hawaiian Mixed Females    0
Length: 78, dtype: int64

Overall, the data is pretty clean and seems easy to work with. I'll play around with it more and see if I can plot anything interesting. 